### PR TITLE
[SYCL][NATIVECPU] Move Native CPU adapter to UR.

### PR DESCRIPTION
**Pull request by martygrant**
UR PR - https://github.com/oneapi-src/unified-runtime/pull/1059

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -238,3 +199,7 @@ endif()
 if ("opencl" IN_LIST SYCL_ENABLE_PLUGINS)
     add_dependencies(sycl-runtime-libraries ur_adapter_opencl)
 endif()
+
+if ("opencl" IN_LIST SYCL_ENABLE_PLUGINS)
```
Comment (fabiomestre) [inline sycl/plugins/unified_runtime/CMakeLists.txt:203]: Should be "native_cpu"

File: sycl/plugins/unified_runtime/ur/adapters/native_cpu/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Cuda adapter
```
Comment (fabiomestre) [inline sycl/plugins/unified_runtime/ur/adapters/native_cpu/README.md:1]: Should be Native CPU adapter

File: sycl/plugins/unified_runtime/CMakeLists.txt (hunk 1 above)
Comment (martygrant) [reply sycl/plugins/unified_runtime/CMakeLists.txt]: Thanks updated now.

File: sycl/plugins/unified_runtime/ur/adapters/native_cpu/README.md (hunk 2 above)
Comment (martygrant) [reply sycl/plugins/unified_runtime/ur/adapters/native_cpu/README.md]: Thanks updated now.

**Review by PietroGhg — APPROVED** [review]
Native CPU LGTM, thank you

**Review by isaacault — APPROVED** [review]
`sycl/plugins/unified_runtime/ur/adapters/native_cpu/image.cpp` LGTM

**Review by kbenzie — APPROVED** [review]

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 3]
```diff
@@ -53,14 +55,16 @@ endif()
...
   # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
-  # Date:   Wed Nov 8 13:32:46 2023 +0000
-  #     Merge pull request #1022 from 0x12CC/l0_usm_error_checking_2
-  #     [UR][L0] Propagate OOM errors from `USMAllocationMakeResident`
-  set(UNIFIED_RUNTIME_TAG ec7982bac6cb3a6b9ed610cd6b7cb41fcbc780dc)
+  # Date:   Tue Nov 7 14:49:55 2023 +0000
+  #     Merge pull request #1026 from kbenzie/benie/test-conformance-sycl-library-dir
+  #     [CTS] Add UR_SYCL_LIBRARY_DIR CMake variable
+  # TODO: Don't merge yet; testing
+  set(UNIFIED_RUNTIME_TAG martin/moveNativeCPUAdapterToUR)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:67]: Now that  Please update https://github.com/oneapi-src/unified-runtime/pull/1059 has merged please pull in the latest changes from the sycl branch and update the `UNIFIED_RUNTIME_REPO` and `UNIFIED_RUNTIME_TAG` as follows.

```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit 659d3f469faa99a886fa680a3d6d20449b109578
  # Merge: 192e9404 f94550b4
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Tue Nov 14 16:45:24 2023 +0000
  #     Merge pull request #1059 from martygrant/martin/moveNativeCPUAdapterToUR
  #     [NATIVECPU] Move Native CPU adapter to UR.
  set(UNIFIED_RUNTIME_TAG 659d3f469faa99a886fa680a3d6d20449b109578)
```

**Review by kbenzie — CHANGES_REQUESTED** [review]

**Review by kbenzie — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers this is ready to merge.

**Pull request merged**